From fa1572f3ea62eabb48e5ada39e5a7a9bd4df044f Mon Sep 17 00:00:00 2001
From: Valentine Barshak <valentine.barshak@cogentembedded.com>
Date: Thu, 8 Sep 2016 04:20:16 +0300
Subject: [PATCH 06/24] ARM64: dts: r8a7795-salvator-x-view: Add gpio-keys
 device node

This adds gpio-keys device which binds SW20-SW22 buttons
to the GPIO keyboard driver.

Signed-off-by: Valentine Barshak <valentine.barshak@cogentembedded.com>
---
 .../boot/dts/renesas/r8a7795-salvator-x-view.dts   | 27 ++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-view.dts b/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-view.dts
index 86dee2c..e996303 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-view.dts
+++ b/arch/arm64/boot/dts/renesas/r8a7795-salvator-x-view.dts
@@ -35,6 +35,7 @@
 /dts-v1/;
 #include "r8a7795.dtsi"
 #include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
 
 / {
 	model = "Renesas Salvator-X.View board based on r8a7795";
@@ -402,6 +403,32 @@
 		#clock-cells = <0>;
 		clock-frequency = <66666666>;
 	};
+
+	gpio_keys {
+		compatible = "gpio-keys";
+
+		button@1 {
+			linux,code = <KEY_1>;
+			label = "SW20";
+			wakeup-source;
+			debounce-interval = <20>;
+			gpios = <&gpio6 11 GPIO_ACTIVE_LOW>;
+		};
+		button@2 {
+			linux,code = <KEY_2>;
+			label = "SW21";
+			wakeup-source;
+			debounce-interval = <20>;
+			gpios = <&gpio6 12 GPIO_ACTIVE_LOW>;
+		};
+		button@3 {
+			linux,code = <KEY_3>;
+			label = "SW22";
+			wakeup-source;
+			debounce-interval = <20>;
+			gpios = <&gpio6 13 GPIO_ACTIVE_LOW>;
+		};
+	};
 };
 
 &pwm1 {
-- 
1.9.3

